app/selectors/custom_status.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {GlobalState} from '@mm-redux/types/store';
+import {UserCustomStatus} from '@mm-redux/types/users';
+
+import {createSelector} from 'reselect';
+import {getCurrentUser, getUser} from '@mm-redux/selectors/entities/users';
+import {get} from '@mm-redux/selectors/entities/preferences';
+import {Preferences} from '@mm-redux/constants';
+import {getConfig} from '@mm-redux/selectors/entities/general';
+
+export function makeGetCustomStatus(): (state: GlobalState, userID?: string) => UserCustomStatus {
+    return createSelector(
+        (state: GlobalState, userID?: string) => (userID ? getUser(state, userID) : getCurrentUser(state)),
+        (user) => {
+            const userProps = user?.props || {};
+            return userProps.customStatus && JSON.parse(userProps.customStatus);
+        },
+    );
+}
+
+export const getRecentCustomStatuses = createSelector(
+    (state: GlobalState) => get(state, Preferences.CATEGORY_CUSTOM_STATUS, Preferences.NAME_RECENT_CUSTOM_STATUSES),
+    (value) => {
+        return value ? JSON.parse(value) : [];
```

Comment:
how can you ensure that `JSON.parse(value)` will return an array?

Reply:
Due to the implementation of adding recent custom statuses in the mattermost-server 
https://github.com/mattermost/mattermost-server/blob/df695115be82e09c27d9b54de754889599b0bf20/app/status.go#L402

Reply:
Yeah but at the very minimum you should be wrapping this in a try/catch as JSON.parse can throw, if it does and is unhandled the app will crash